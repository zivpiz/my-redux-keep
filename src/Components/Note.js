import React from 'react';
import {connect} from 'react-redux';
import {changeCheckedNote, changeUncheckedNote} from "../actions";
import './../style.css';


const handleChange = (props) => {
    if (props.checked)
        props.changeCheckedNote(props);
    else props.changeUncheckedNote(props);
};

const Note = (props) => {
    return (
        <div className="Note">
            <input type="checkbox" className="NoteCheckbox" onChange={() => handleChange(props)}
                   checked={props.checked}/>
            <label className={`NoteTextChecked-${props.checked}`}>{props.text}</label>
        </div>
    )
};

export default connect(null, {changeCheckedNote, changeUncheckedNote})(Note);