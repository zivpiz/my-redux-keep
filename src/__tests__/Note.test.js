import Note from "../Components/Note";
import React from 'react';

describe("Test Note Component ", ()=>{

    let wrapper;
    const onChangeMock = jest.fn();

    beforeEach(()=>{
        wrapper = shallow(<Note text="Note Test" checked={true}
                                handleCheck={onChangeMock}/>);
    });

    test('Renders appropriate snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Calls onChangeMock when checkbox is clicked', () => {
        wrapper.find('input').simulate('change');
        expect(onChangeMock).toHaveBeenCalled();
    });

    test('Names label according to checked status', () => {
        const label = wrapper.find('label');
        expect(label.props().className).toBe("NoteTextChecked-true");
    });
});


