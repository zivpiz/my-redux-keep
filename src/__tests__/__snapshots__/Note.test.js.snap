// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Note Component  renders appropriate snapshot 1`] = `
<div
  className="Note"
>
  <input
    className="NoteCheckbox"
    defaultChecked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="NoteTexttrue"
  >
    Note Test
  </label>
</div>
`;
